File: Data Structures & Algorithms/longest-palindromic-substring/submission-4.py
class Solution:
    def longestPalindrome(self, s: str) -> str:

        n = len(s)
        res =""

        def expand(i,j):
            while(i>=0 and j<n and s[i]==s[j]):
                i-=1
                j+=1
            return s[i+1:j]

        for i in range(n):
            left = expand(i,i)
            right = expand(i,i+1)

            if len(left)>len(res):
                res = left
            if len(right)>len(res):
                res = right
        
        return res

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-4.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> res;
        if (root==nullptr){
            return res;
        }        
        
        queue<TreeNode*> nodes;
        nodes.push(root);

        while(!nodes.empty()){
            vector<int> level;
            int length = nodes.size();
            for (int i =0;i<length;i++){
                TreeNode* node = nodes.front();
                nodes.pop();
                level.push_back(node->val);

                if (node->left!= nullptr){
                    nodes.push(node->left);
                }
                if (node->right!= nullptr){
                    nodes.push(node->right);
                }

            }
            res.push_back(level);
        }

        return res;
        
    }
};
